lib/translator/read.cc

```diff
@@ -0,0 +1,183 @@
+#include "read.h"
+
+#include "absl/types/span.h"
+
+#include "common.h"
+
+namespace translator {
+
+namespace {  // anonymous namespace for helper functions
+
+// Section 5.3
+// https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf
+StatusCode BuildPrinfo(uint8_t rdprotect, uint8_t& prinfo) {
```

Comment:
suggestion: we can move this function in common.h since write will also be using this

Reply:
Sounds good. I'll also open that change in a new PR

Reply:
update: Read and Write have different logic when building prinfo, so it is not desirable to abstract this out for both read and write